nemo/collections/multimodal/speech_llm/models/modular_t5_models.py

```diff
@@ -923,25 +985,42 @@ def inference_step(self, dataloader_iter, mode, dataloader_idx=0):
         return outputs
 
     def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Any:
+        # the following supports STT (audio-text) inference
 
         batch = move_to_device(batch, device=self.device)
-        encoder_input, attention_mask, enc_mask = self.prepare_llm_input(batch)
-        # enc_input = speech and text prompt
-        # dec_input and label = text output label
-        predicted_token_ids, log_probs = self.frozen_model.decode(
-            tokens_enc=None,
-            enc_mask=enc_mask,
-            num_tokens_to_generate=self._inference_config['tokens_to_generate'],
-            encoder_input=encoder_input,
-            tokenizer=self.tokenizer,
-            bos_id=self.bos_id,
-        )
+        audio_batch = {k: v for k, v in batch.items() if not k.startswith("text_")}
+        text_batch = {k: v for k, v in batch.items() if k.startswith("text_")}
+        assert (
+            audio_batch or text_batch and not (audio_batch and text_batch)
+        ), f"Expecting only text or audio batch, got {len(text_batch)=} and {len(audio_batch)=}"
+
+        if 'audio_signal' in audio_batch:
+            input_text = audio_batch['contexts']
+            labels = audio_batch['answers']
+            encoder_input, attention_mask, enc_mask = self.prepare_llm_input(audio_batch)
+            predicted_token_ids, log_probs = self.frozen_model.decode(
+                tokens_enc=None,
+                enc_mask=enc_mask,
+                num_tokens_to_generate=self._inference_config['tokens_to_generate'],
+                encoder_input=encoder_input,
+                tokenizer=self.tokenizer,
+                bos_id=self.bos_id,
+            )
+        if text_batch:
```

Comment:
assert only one of audio_batch and text_batch can exisit

Reply:
a few lines above there is 
```
        assert (
            audio_batch or text_batch and not (audio_batch and text_batch)
        ), f"Expecting only text or audio batch, got {len(text_batch)=} and {len(audio_batch)=}"
```